utils/rocvideodecode/roc_video_dec.cpp

```diff
@@ -39,9 +39,10 @@ RocVideoDecoder::RocVideoDecoder(int device_id, OutputSurfaceMemoryType out_mem_
     // create rocdec videoparser
     RocdecParserParams parser_params = {};
     parser_params.codec_type = codec_id_;
-    parser_params.max_num_decode_surfaces = 1;
+    // Note this is tunable
+    parser_params.max_num_decode_surfaces = MAX_DPB_FRAMES + DISPLAY_DELAY;
```

Comment:
The current change adjusts the buffer pool to 18, resulting in additional unused memory being allocated per stream. Prior to this pull request, 7 surfaces were allocated to decode the AMD_driving_virtual_20-H265.mp4 sample. However, after the pull request, we are now allocating 18 surfaces, which represents an additional overhead.



**The current ToT develop branch:**
./videodecode -i ../../../data/videos/AMD_driving_virtual_20-H265.mp4 
info: Input file: AMD_driving_virtual_20-H265.mp4
info: Using GPU device 0 - AMD Radeon RX 6750 XT[gfx1031] on PCI bus 0b:00.0
info: decoding started, please wait!
Input Video Information
        Codec        : H.265/HEVC
        Frame rate   : 30000/1001 = 29.97 fps
        Sequence     : Progressive
        Coded size   : [2048, 1024]
        Display area : [0, 0, 2048, 1024]
        Chroma       : YUV 420
        Bit depth    : 8
Video Decoding Params:
        **Num Surfaces : 7**
        Crop         : [0, 0, 2048, 1024]
        Resize       : 2048x1024

info: Total frame decoded: 601
info: avg decoding time per frame: 1.79221 ms
info: avg FPS: 557.972

**The current PR:**
./videodecode -i ../../../data/videos/AMD_driving_virtual_20-H265.mp4 
info: Input file: AMD_driving_virtual_20-H265.mp4
info: Using GPU device 0 - AMD Radeon RX 6750 XT[gfx1031] on PCI bus 0b:00.0
info: decoding started, please wait!
Input Video Information
        Codec        : H.265/HEVC
        Frame rate   : 30000/1001 = 29.97 fps
        Sequence     : Progressive
        Coded size   : [2048, 1024]
        Display area : [0, 0, 2048, 1024]
        Chroma       : YUV 420
        Bit depth    : 8
Video Decoding Params:
        **Num Surfaces : 18**
        Crop         : [0, 0, 2048, 1024]
        Resize       : 2048x1024

info: Total frame decoded: 601
info: avg decoding time per frame: 1.61563 ms
info: avg FPS: 618.955

Reply:
Right. The size 18 of the decode buffer pool is set by the user. In our case, the user is the util layer. It can be tuned. I have a suggestion that we can discuss in the review meeting.

Reply:
Why should the user decide on the decode buffer pool? It should be determined by the parser.

Reply:
Because the user can determine how much buffering it wants before display. Note decode buffer pool is not the same as DPB.

Reply:
We probably want to use the display delay as the tuning parameter. Which is my suggestion.

Reply:
Briefly, the suggestion is to set the decode buffer pool size to DPB size + display delay. The current hard coded 18 only considers the worst scenario.

Reply:
Commit 2ad9e01 should address this issue. I'll push another commit for other small changes.